File: Python/problem2.py
import bisect 
from typing import List
class Solution(object):
    def minPatches(self, nums, n):
        patches = 0
        i = 0
        acum = 1 #pk iniciamos en 1
        while acum <= n:
            if i < len(nums) and nums[i] <= acum:#Eran separados pero pro esto funciono idk
                acum += nums[i]#aumenta lo que equivalga el numero
                #print('acum ',acum)
                i+=1
                #print('i ',i)
            else:
                patches += 1 #Si no entra quieer decir que ocupamos un patch
                #print('patches ',patches)
                acum += acum #aumenta en si pk vamos por el sigueinte nummero y ya teneindo en cuenta el patch     
                #print('acum2 ',acum)

        return patches
        
        

        

File: Java/problem2.java
class Solution {
    public int minPatches(int[] nums, int n) {
        int patches = 0;
        int i = 0;
        //int acum = 1;
        //List<Integer> nums=new ArrayList<Integer>();
        //System.out.print(patches+"_"+i);
        //while (acum <= n) {
        for(long acum = 1;acum<=n;){
            if (i < nums.length && nums[i] <= acum) {
                acum += nums[i];
                i ++;
            }else{
                patches ++;
                acum += acum;
            }
        }
        return patches;
    }
}